consumer/git-publisher/src/test/java/org/arquillian/pact/consumer/publisher/git/GitPactPublisherTest.java

```diff
@@ -0,0 +1,196 @@
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class GitPactPublisherTest {
+
+    @Mock
+    GitOperations gitOperations;
+
+    @Mock
+    Git git;
+
+    @Mock
+    Repository repository;
+
+    @Mock
+    PullResult pullResult;
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+
+    @Test
+    public void should_clone_and_copy_pact_files_and_push() throws IOException {
+
+        byte[] content = "Contract File".getBytes();
+        final File pactLocation = temporaryFolder.newFolder("input");
+        Files.copy(new ByteArrayInputStream(content),
+                new File(pactLocation, "pact.txt").toPath());
+
+        GitPactPublisher gitPactPublisher = new GitPactPublisher();
+        gitPactPublisher.gitOperations = gitOperations;
+
+        File repository = temporaryFolder.newFolder("repo");
+        Map<String, Object> config = new HashMap<>();
+        config.put(GitPactPublisher.URL, "myurl");
+        config.put(GitPactPublisher.COMMENT, "my comment");
+        config.put(GitPactPublisher.REPOSITORY, repository.getAbsolutePath());
+        gitPactPublisher.configure(config);
+
+        when(gitOperations.isValidGitRepository(Paths.get(repository.getAbsolutePath()))).thenReturn(false);
+        when(gitOperations.cloneRepository("myurl", Paths.get(repository.getAbsolutePath()))).thenReturn(git);
+        when(gitOperations.isLocalBranch(git, "master")).thenReturn(true);
+
+        when(git.getRepository()).thenReturn(this.repository);
+        when(this.repository.getDirectory()).thenReturn(new File(repository, ".git"));
+
+
+        gitPactPublisher.store(Paths.get(pactLocation.getAbsolutePath()));
+        verify(gitOperations).addAndCommit(git, "my comment");
+        verify(gitOperations).pushRepository(git, "origin");
+
+        assertThat(new File(repository, "pact.txt")).exists().hasContent("Contract File");
+
+    }
+
+    @Test
+    public void should_reuse_git_repo_copy_pact_files_and_push() throws IOException {
+
+        byte[] content = "Contract File".getBytes();
+        final File pactLocation = temporaryFolder.newFolder("input");
+        Files.copy(new ByteArrayInputStream(content),
+                new File(pactLocation, "pact.txt").toPath());
+
+        GitPactPublisher gitPactPublisher = new GitPactPublisher();
+        gitPactPublisher.gitOperations = gitOperations;
+
+        File repository = temporaryFolder.newFolder("repo");
+        Map<String, Object> config = new HashMap<>();
+        config.put(GitPactPublisher.URL, "myurl");
+        config.put(GitPactPublisher.COMMENT, "my comment");
+        config.put(GitPactPublisher.REPOSITORY, repository.getAbsolutePath());
+        gitPactPublisher.configure(config);
```

Comment:
We have quite a bit of verbose and duplicated code in those test methods. How about making some meaningful method / builder to make tests more readable?